tests/backup/backup_resiliency_test.go

```diff
@@ -1312,10 +1312,12 @@ var _ = Describe("{RebootNodesWhenBackupsAreInProgress}", func() {
 		log.Infof("Deleting the deployed applications on application cluster")
 		opts := make(map[string]bool)
 		opts[SkipClusterScopedObjects] = true
-		ValidateAndDestroy(contexts, opts)
+		DestroyApps(contexts, opts)
```

Comment:
Why do you want to skip validation, skipping validation might lead to missing the bug.
Can you check why postgres is not up?

Reply:
@lsrinivas-pure What's the point of validating if the app is running in `JustAfterEach()`? The problem with the current implementation is that once it fails in `ValidateAndDestroy ` it doesn't reset the context causing the subsequent  tests to fails. 

The postgres pods are failing as the they are unable to find the pgbench database. we aren't sure if this issue is something coming from the portworx side of things. 

Describe of the pod:
```
Events:
  Type     Reason     Age                     From     Message
  ----     ------     ----                    ----     -------
  Warning  Unhealthy  15m (x798 over 17h)     kubelet  Liveness probe failed: psql: FATAL:  database "pgbench" does not exist
  Warning  BackOff    5m34s (x3450 over 17h)  kubelet  Back-off restarting failed container
  Warning  Unhealthy  27s (x1922 over 17h)    kubelet  Readiness probe failed: psql: FATAL:  database "pgbench" does not exist
```

logs of the pods: 
```
[root@ip-10-13-201-90 ~]# kubectl logs -n postgres-backup-pxbackuptask-0-05-07-12h53m58s  postgres-646f6f7487-8vfcm

PostgreSQL Database directory appears to contain a database; Skipping initialization

LOG:  could not create IPv6 socket: Address family not supported by protocol
LOG:  database system was shut down at 2023-05-08 04:39:16 UTC
LOG:  MultiXact member wraparound protections are now enabled
LOG:  database system is ready to accept connections
LOG:  autovacuum launcher started
FATAL:  database "pgbench" does not exist
FATAL:  database "pgbench" does not exist
FATAL:  database "pgbench" does not exist
FATAL:  database "pgbench" does not exist
FATAL:  database "pgbench" does not exist
FATAL:  database "pgbench" does not exist
FATAL:  database "pgbench" does not exist
FATAL:  database "pgbench" does not exist
FATAL:  database "pgbench" does not exist
FATAL:  database "pgbench" does not exist
LOG:  received fast shutdown request
LOG:  aborting any active transactions
LOG:  autovacuum launcher shutting down
LOG:  shutting down
LOG:  database system is shut down
```

Reply:
I believe, it's better to split and call this in two places 
1. ValidateApplications() - > Should call right after the node reboot . So that will get to know if there is any issue after node reboot.
2. DestroyApps() -> Lets keep this in teardown part

Reply:
This might be out of scope for this PR. To ensure that errors are properly handled, it would be appropriate to implement an errChan parameter in the ValidateAndDestroy function. This will enable the channel to be updated whenever errors occur.

Reply:
@kshithijiyer-px 
Ideally validating the apps should be in your test itself.
I would suggest you move this validation to inside the test fail the test if any validation fails and since context switch is happening in aftereach it should work

Reply:
@lsrinivas-pure We are validating the deployed app in our test case itself. We have this step
```
Step("Validating the deployed applications", func() {
			log.InfoD("Validating the deployed applications on destination cluster")
			ValidateApplications(contexts)
		})
```
The test case actually failed in this step itself. So now we are asking if it is necessary to validate again in the `JustAfterEach` block again because it will never pass the validation here.

Reply:
So we have `ValidateApplications` inside the test, so I think it is safe to use `DestroyApps` in the cleanup part, right?

Reply:
@lsrinivas-pure Waiting for your final approval.

Reply:
as per discussion with @kshithijiyer-px this change of moving validations from AfterEach to test will be taken care by @snigdha-px in a separate PR

@snigdha-px please mention that task link in this PR

Reply:
Yes @lsrinivas-pure we have created a bug to handle this state where the application is up & running or not after reboot the nodes. With this approach, app state validation goes to testcase and app deletion will be in teardown.
https://portworx.atlassian.net/jira/software/c/projects/PA/issues/PA-1018